core/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunction.java

```diff
@@ -49,6 +49,16 @@
      *            the value which decay function should take once the distance
      *            reaches this scale
      * */
-    public double processScale(double scale, double decay);
+    public abstract double processScale(double scale, double decay);
 
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
the function doesn't hold instance members, but comparing that the function is the same class is enough here. otherwise we rely on reference equality which is not what we need.